scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/SubmitWindow.java

```diff
@@ -1140,12 +1145,33 @@ private void updateVariables(JSONObject updatedVariablesJson) {
         }
     }
 
+    private boolean matchesCopyPattern(String variableName) {
+        FormItem matchField = Arrays.stream(fields)
+                                    .filter(field -> field.getName().equals(variableName))
+                                    .findFirst()
+                                    .orElse(null);
+        if (matchField != null && matchField.getValue() != null) {
+            String matchVariableName = extractVariableName(matchField.getValue().toString());
+            return matchVariableName != null;
+        }
+        return false;
+    }
+
+    private String extractVariableName(String variablePatternCheck) {
+        return variables.keySet()
+                        .stream()
+                        .filter(variableName -> variablePatternCheck.equals("$" + variableName) ||
```

Comment:
you should use contains and not equals

i.e. if value is: `something_${another_variable}` it should not be replaced